### PR TITLE
Allow multi-use invite codes

### PR DESCRIPTION
**Pull request by ankonym**
This will allow admins to generate invite codes and specify the number of times they can be used to create new accounts. (If no number if specified, single-use invite codes will be generated.) 

This is done by introducing a max_uses property and comparing the invite count to the maximum number of uses, and adding an optional parameter to the rake task.

### CODE REVIEW

**Comment by azul** [comment]
@ankonym looks clean and good. :sparkles: . I have two minor things i will comment on. Please take a quick look - maybe they can still be tweaked.

File: lib/tasks/invite_code.rake [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
+task :generate_invites, [:n, :u] => :environment do |task, args|
 
-    codes = args.n
-    codes = codes.to_i
+  codes = args.n
+  codes = codes.to_i
 
-    codes.times do |x|
+  if args.u == nil
+    max_uses = 1
```
Comment (azul) [inline lib/tasks/invite_code.rake]: We're already setting the default in the model. I don't think we will ever want to change it. But maybe we can only overwrite it below if it was handed as an arg?

File: lib/tasks/invite_code.rake [hunk 2]
```diff
@@ -1,16 +1,25 @@
...
+
+  elsif
+    max_uses = args.u
+    max_uses = max_uses.to_i
```
Comment (azul) [inline lib/tasks/invite_code.rake]: The codes.to_i above is needed to the codes.times below works. (times is only defined on integers and not on strings). :thought_balloon: But do we also need it for the max_uses setter? :confused: :grey_question: 
It sure does not do no harm - but maybe we can cut a tiny bit of compexity. :scissors:

File: lib/tasks/invite_code.rake (hunk 1 above)
Comment (ankonym) [inline lib/tasks/invite_code.rake]: Ah, I've removed that (and made the "elsif" the "if" check to make it a bit shorter). Thanks for the comment!

File: lib/tasks/invite_code.rake (hunk 2 above)
Comment (ankonym) [inline lib/tasks/invite_code.rake]: We actually didn't need it, I just stole some code from a few lines above there. :) 
Cleaned up now!

File: lib/tasks/invite_code.rake [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
+  codes = codes.to_i
 
-    codes.times do |x|
+  if args.u != nil
+    max_uses = args.u
+  end
+
+  codes.times do |x|
     x = InviteCode.new
+    x.max_uses = max_uses
```
Comment (azul) [inline lib/tasks/invite_code.rake]: does this work if max_users is not defined - i.e. if you don't supply an argument for u?

Comment (ankonym) [inline lib/tasks/invite_code.rake]: Yes, if only n is given, the max uses will automatically be set to 1. 
I also included that in the documentation. :bowtie:

bildschirmfoto 2015-10-13 um 17 03 50

Comment (azul) [inline lib/tasks/invite_code.rake]: perfect :100:

**Pull request merged**